libcontainer/cgroups/fs2/memory.go

```diff
@@ -50,8 +50,10 @@ func setMemory(dirPath string, cgroup *configs.Cgroup) error {
 		// memory and memorySwap set to the same value -- disable swap
 		swapStr = "0"
 	}
-	if err := fscommon.WriteFile(dirPath, "memory.swap.max", swapStr); err != nil {
-		return err
+	if swapStr != "" {
```

Comment:
CI error:
```
# time="2020-05-10T01:05:29Z" level=error msg="container_linux.go:348: starting container process caused: process_linux.go:438: container init caused: process_linux.go:404: setting cgroup config for procHooks process caused: failed to write \"\" to \"/sys/fs/cgroup/runc-cgroups-integration-test/test-cgroup/memory.swap.max\": open /sys/fs/cgroup/runc-cgroups-integration-test/test-cgroup/memory.swap.max: permission denied"
```
This error is introduced by #2370 . So we need to fix it in here.

Reply:
This is a separate fix and it requires a separate patch.

Also, why haven't we caught this before?

Reply:
Can we have this fix as a separate patch? And I still don't understand why we haven't caught it before.

Reply:
We don’t check memorySwap==0

Reply:
@kolyshkin Do we still need to submit a separate patch?

Reply:
The seprate patch: #2405

Reply:
This part is fixed by #2410